cli/internal/cmd/apply.go

```diff
@@ -486,29 +495,33 @@ func (a *applyCmd) validateInputs(cmd *cobra.Command, configFetcher attestationc
 
 	// Validate microservice version (helm versions) in the user's config matches the version of the CLI
 	// This makes sure we catch potential errors early, not just after we already ran Terraform migrations or the init RPC
-	if !a.flags.force && !a.flags.skipPhases.contains(skipHelmPhase) && !a.flags.skipPhases.contains(skipInitPhase) {
+	if !a.flags.force && !a.flags.skipPhases.contains(skipHelmPhase, skipInitPhase) {
 		if err := validateCLIandConstellationVersionAreEqual(constants.BinaryVersion(), conf.Image, conf.MicroserviceVersion); err != nil {
 			return nil, nil, err
 		}
 	}
 
-	// Constellation on QEMU or OpenStack don't support upgrades
-	// If using one of those providers, make sure the command is only used to initialize a cluster
-	if !(conf.GetProvider() == cloudprovider.AWS || conf.GetProvider() == cloudprovider.Azure || conf.GetProvider() == cloudprovider.GCP) {
-		if a.flags.skipPhases.contains(skipInitPhase) {
-			return nil, nil, fmt.Errorf("upgrades are not supported for provider %s", conf.GetProvider())
-		}
-		// Skip Terraform phase
-		a.log.Debugf("Skipping Infrastructure upgrade")
-		a.flags.skipPhases.add(skipInfrastructurePhase)
+	// Constellation does not support image upgrades on all CSPs. Not supported are: QEMU, OpenStack
+	// If using one of those providers, print a warning when trying to upgrade the image
+	if !(conf.GetProvider() == cloudprovider.AWS || conf.GetProvider() == cloudprovider.Azure || conf.GetProvider() == cloudprovider.GCP) &&
+		!a.flags.skipPhases.contains(skipImagePhase) {
+		cmd.PrintErrf("Image upgrades are not supported for provider %s\n", conf.GetProvider())
+		cmd.PrintErrln("Image phase will be skipped")
+		a.flags.skipPhases.add(skipImagePhase)
 	}
 
 	// Check if Terraform state exists
+	// If it does not, assume we are creating a new cluster,
```

Comment:
we cannot make this assumption with the Terraform modules

Reply:
We have to.
Self-managed infrastructure users should use `--skip-phases=infrastructure`